architecture/adr-list/adr-001-cheqd-did-method.md

```diff
@@ -0,0 +1,616 @@
+
+Usage of UUID-style identifiers significantly simplifies the generation and implementation of unique identifiers, [with extremely-low probability of a collission where the same UUID](http://www.h2database.com/html/advanced.html#uuid) is generated independently.
+
+Any client application can generate these UUIDs using their own preferred implementation in any programming language, as opposed to the method-specific logic required for Indy-style DID identifiers.
+
+#### Indy-style Unique Identifiers
+
+Alternatively, [the `unique-id` **may** also be generated similar to the `did:indy method`](https://hyperledger.github.io/indy-did-method/#indy-did-method-identifiers) from the initial public key of the DID (e.g., base58 encoding of the first 16 bytes of the SHA256 of the first Verification Method `Ed25519` public key). This `unique-id` format is referred to as the "Indy-style" unique identifier in our documentation.
+
+Support for Indy-style unique identifiers makes compatibility with Indy-based client SDKs, such as those based on [Hyperledger Aries](https://www.hyperledger.org/use/aries).
+
+#### Namespace
+
+If no `namespace` is specified, it assumed to be default `namespace` for the network/ledger the request is targetted at. This will generally be `mainnet` for the primary production cheqd network.
+
+### Syntax for `did:cheqd` method
+
+The cheqd DID method ABNF to conform with [DID syntax guidelines](https://www.w3.org/TR/did-core/#did-syntax) is as follows:
+
+```abnf
+cheqd-did       = "did:cheqd:" [namespace]
+namespace       = 1*namespace-char ":" unique-id
+namespace-char  = ALPHA / DIGIT
+unique-id       = *id-char / UUID
+id-char         = ALPHA / DIGIT
+```
+
+>**Note:** The `*id-char unique-id` must be 16 bytes of Indy-style base58 encoded identifier.
+
+#### ABNF syntax for UUID-style identifiers
+
+Where the [formal definition of UUIDs is represented using this ABNF](https://datatracker.ietf.org/doc/html/rfc4122#section-3):
+
+```abnf
+UUID                   = time-low "-" time-mid "-"
+                         time-high-and-version "-"
+                         clock-seq-and-reserved
+                         clock-seq-low "-" node
+time-low               = 4hexOctet
+time-mid               = 2hexOctet
+time-high-and-version  = 2hexOctet
+clock-seq-and-reserved = hexOctet
+clock-seq-low          = hexOctet
+node                   = 6hexOctet
+hexOctet               = hexDigit hexDigit
+hexDigit =
+      "0" / "1" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9" /
+      "a" / "b" / "c" / "d" / "e" / "f" /
+      "A" / "B" / "C" / "D" / "E" / "F"
+```
+
+### Examples of `did:cheqd` identifiers
+
+#### Indy-style
+
+A DID written to the cheqd "mainnet" ledger `namespace` with a 22-character Indy-style identifier:
+
+```text
+did:cheqd:mainnet:TAwT8WVt3dz2DBAifwuSkn
+```
+
+A DID written to the cheqd "testnet" ledger `namespace` with a 22-character Indy-style identifier:
```

Comment:
DID should be 16 bytes of Indy-style base58 encoded identifier.
16 bytes of Indy style testnet identifier: `did:cheqd:testnet:VwZmDHc9vHtJe8W3wofzQk`